glue/core/tests/test_roi.py

```diff
@@ -340,6 +341,36 @@ def test_str(self):
         assert type(str(self.roi)) == str
 
 
+class TestCategorical(object):
+
+    def setup_method(self, method):
+        self.roi = CategoricalRoi()
+
+    def test_defined(self):
+
+        nroi = CategoricalRoi(categories=['a', 'b', 'c'])
+        assert nroi.defined()
+        nroi.reset()
+        assert not nroi.defined()
+
+    def test_loads_from_components(self):
+
+        comp = CategoricalComponent(np.array(['a', 'a', 'b']))
+        self.roi.update_categories(comp)
+
+        np.testing.assert_array_equal(self.roi.categories,
+                                      np.array(['a', 'b']))
+
+    def test_applies_components(self):
+
+        comp = CategoricalComponent(np.array(['a', 'b', 'c']))
+        self.roi.update_categories(CategoricalComponent(np.array(['a', 'b'])))
+        contained = self.roi.contains(comp, None)
+        np.testing.assert_array_equal(contained,
+                                      np.array([True, True, False]))
+
+
+
```

Comment:
If I remember the problem correctly, the issue is that CategoricalComponents are internally represented as integer arrays, with a mapping from index to category. This mapping was potentially different for two Components, even if their categories overlapped / were the same. Because the ROI logic was using the underlying numerical arrays and not the categories, they in general wouldn't filter properly.

Do I have that right? If so, let's add some tests that setup up multiple CategoricalCoponents with different number->category mappings, and ensure they filter correctly.